From kernel-bounces+gregkh=suse.de@suse.de  Thu Sep 11 02:14:58 2008
From: Hannes Reinecke <hare@suse.de>
Date: Wed, 10 Sep 2008 14:16:16 +0200
Subject: Driver core: Remove suspend/resume callbacks for device type
To: kernel@suse.de
Message-ID: <20080910121616.7C34D18C734@pentland.suse.de>

struct device_type accumulated suspend/resume callbacks which are
unused. Kill them.

Signed-off-by: Hannes Reinecke <hare@suse.de>
Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/power/main.c |   25 -------------------------
 include/linux/device.h    |    3 ---
 2 files changed, 28 deletions(-)

--- a/drivers/base/power/main.c
+++ b/drivers/base/power/main.c
@@ -363,18 +363,6 @@ static int resume_device(struct device *
 			goto End;
 	}
 
-	if (dev->type) {
-		if (dev->type->pm) {
-			pm_dev_dbg(dev, state, "type ");
-			error = pm_op(dev, dev->type->pm, state);
-		} else if (dev->type->resume) {
-			pm_dev_dbg(dev, state, "legacy type ");
-			error = dev->type->resume(dev);
-		}
-		if (error)
-			goto End;
-	}
-
 	if (dev->class) {
 		if (dev->class->pm) {
 			pm_dev_dbg(dev, state, "class ");
@@ -608,19 +596,6 @@ static int suspend_device(struct device 
 			goto End;
 	}
 
-	if (dev->type) {
-		if (dev->type->pm) {
-			pm_dev_dbg(dev, state, "type ");
-			error = pm_op(dev, dev->type->pm, state);
-		} else if (dev->type->suspend) {
-			pm_dev_dbg(dev, state, "legacy type ");
-			error = dev->type->suspend(dev, state);
-			suspend_report_result(dev->type->suspend, error);
-		}
-		if (error)
-			goto End;
-	}
-
 	if (dev->bus) {
 		if (dev->bus->pm) {
 			pm_dev_dbg(dev, state, "");
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -273,9 +273,6 @@ struct device_type {
 	int (*uevent)(struct device *dev, struct kobj_uevent_env *env);
 	void (*release)(struct device *dev);
 
-	int (*suspend)(struct device *dev, pm_message_t state);
-	int (*resume)(struct device *dev);
-
 	struct pm_ops *pm;
 };
 
